controllers/workspace/devworkspace_controller.go

```diff
@@ -368,6 +532,53 @@ func (r *DevWorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		devfilePodAdditions.InitContainers = append([]corev1.Container{*projectClone}, devfilePodAdditions.InitContainers...)
 	}
 
+	// Inject operator-configured init containers
+	if workspace.Config != nil && workspace.Config.Workspace != nil && len(workspace.Config.Workspace.InitContainers) > 0 {
+		// Check if init-persistent-home should be disabled
+		disableHomeInit := pointer.BoolDeref(workspace.Config.Workspace.PersistUserHome.DisableInitContainer, constants.DefaultDisableHomeInitContainer)
+
+		// Prepare patches: filter and preprocess init containers from config
+		patches := []corev1.Container{}
+		for _, container := range workspace.Config.Workspace.InitContainers {
+			// Special handling for init-persistent-home
+			if container.Name == constants.HomeInitComponentName {
+				// Skip if persistent home is disabled
+				if !home.PersistUserHomeEnabled(workspace) {
+					reqLogger.Info("Skipping init-persistent-home container: persistent home is disabled")
+					continue
+				}
+				// Skip if init container is explicitly disabled
+				if disableHomeInit {
+					reqLogger.Info("Skipping init-persistent-home container: DisableInitContainer is true")
+					continue
+				}
+				// Apply defaults and validation for init-persistent-home
+				container, err = defaultAndValidateHomeInitContainer(container, workspace)
```

Comment:
IMHO I would prefer if we don't apply defaults for the home init container here, because I would think that the defaults are already applied when initially added to `devfilePodAdditions.InitContainers`, is that correct?

Reply:
done